code-challenges.js

```diff
@@ -16,41 +34,183 @@ const fruit2 = "banana"
+
+    // Yes; once i created this  array1.concat(array2).length on like 117 it now returns 10.
+
+    console.log("--- problem 3 ---")
 // --------------------3) Create the code that will reverse the letters of a string using the test variable provided below. Expected output: "2202 atleD"
 
-// Pseudo code:
+// Pseudo code: I will declare a function named reverseString that will take one string input and reverse it then output the value. I may take the string and assign each element to a array then reverse it then 
+
+// actually i might do a for loop and interate through the string assigning the values to an array then reverse it
 
 const currentCohort = "Delta 2022"
+    // var reverseSt = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ,11]
+
+    // const reverseString = (inputA) => {
+    // for (i = 0; i < inputA.length; i++) {
+    //     reverseSt.unshift[inputA[""]]
+        
+    // }
+    // console.log(reverseSt)
+    // }
+    // reverseString(currentCohort)
+
+    // my code was not working so I started looking at my notes
+
+//console.log(currentCohort.split(""))
+
+reverseStr = currentCohort.split("")
+
+ // I had to google how to reverse it but the code works. 
+
+console.log(reverseStr.reverse())
+ 
+   
+
+console.log("--- problem 4 ---")
 
 
 // --------------------4) Create a statement that will evaluate whether each number in an array is even or odd. Use the test variable provided below. Expected output: "odd" "even" "odd" "even" "odd" "even"
 
-// Pseudo code:
+// Pseudo code: I am going to use a for if else statement to iterate through each element ant test if its odd or even with a modulo, if its even I return the string even, if its off i will return the string odd
 
 const myArray = [13, 34, 5, 10, 27, 42]
 
 
+const balanceCheck = (array12) => {
+    for (i = 0; i < array12.length ; i++) {
+        if (array12[i] % 2 === 0) {
+            console.log(array12[i] + " Even")
+        } else {
+            console.log(array12[i] + " Odd")
+        }
+    }
+    return null
+}
+
+balanceCheck(myArray)
+    // for some reason I walways expect just calling a function to output it to the console
+console.log(balanceCheck(myArray))
+    // The first time i tried my only output was "Odd", so I'm going in and adding console.logs to the if statements to see what they output.
+    // Now I'm running into an issue where once the if else statement inside the loop returns a value it ends the loop; i had to change them to output the string to the console each time
+    // My final change was to create console.log(array12[i] + " Even") as one line inside the if statement instead of two seperate console logs
+
+
+    // --- problem 4 ---
+    // 13 Odd
+    // 34 Even
+    // 5 Odd
+    // 10 Even
+    // 27 Odd
+    // 42 Even
+    // 13 Odd
+    // 34 Even
+    // 5 Odd
+    // 10 Even
+    // 27 Odd
+    // 42 Even
+    // undefined
+// I am going to add a return null to the end of the function to stop it from returning undefined in the final code.
```

Comment:
Great work!  Just remember, every function needs a return.  Instead of using console.log in your conditionals, you can return and that should make it so you won't get undefined.